### PR TITLE
[Composite Alerts] prepare Expression to become Optional at times

### PR DESCRIPTION
**Pull request by mattcarmody**
This is part 1 of adding Composite Alerts to the terraform provider. This should be a no-op, it just lays the foundation for upcoming changes.

### CODE REVIEW

File: lightstep/resource_dashboard.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
-func getUnifiedQuerySchema() map[string]*schema.Schema {
+func getUnifiedQuerySchemaMap() map[string]*schema.Schema {
```
Comment (mattcarmody) [inline lightstep/resource_dashboard.go:12]: a bunch of functions are named with the same pattern. some return a schema while others return a map of schemas so I renamed some

File: client/metric_conditions.go [hunk 2]
```diff
@@ -30,9 +30,13 @@ type AlertingRule struct {
 }
 
 type Expression struct {
+	SubAlertExpression
```
Comment (mattcarmody) [inline client/metric_conditions.go:33]: `SubAlertExpression` will be used without `IsMulti` in follow up PR(s)

File: lightstep/resource_metric_condition.go [hunk 3]
```diff
@@ -344,6 +312,66 @@ func getThresholdSchema() map[string]*schema.Schema {
...
+	resource := getCompositeSubAlertExpressionResource()
+	resource.Schema["is_multi"] = getIsMultiSchema()
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MaxItems: 1,
+		MinItems: 1,
+		Elem:     resource,
+	}
+}
```
Comment (mattcarmody) [inline lightstep/resource_metric_condition.go:338]: these are the separate paths of the branch above and they're quite similar, the difference is Required/Optional & MinItems. I think breaking things out this way will make a bit more sense when I add the `SubAlert` version

File: lightstep/resource_metric_condition.go [hunk 4]
```diff
@@ -488,9 +516,7 @@ func (p *resourceUnifiedConditionImp) resourceUnifiedConditionImport(ctx context
 }
 
 func getUnifiedConditionAttributesFromResource(d *schema.ResourceData, schemaType ConditionSchemaType) (*client.UnifiedConditionAttributes, error) {
-	expression := d.Get("expression").([]interface{})[0].(map[string]interface{})
-
-	thresholds, err := buildThresholds(d)
```
Comment (mattcarmody) [inline lightstep/resource_metric_condition.go:493]: `buildThresholds` makes assumptions about `expression` being present and ends up in the `Expression` so I moved it inside `buildExpression`

**Review by gontzess — APPROVED** [review]
looks good, LGTM!

File: lightstep/resource_metric_condition.go [hunk 5]
```diff
@@ -344,6 +312,67 @@ func getThresholdSchema() map[string]*schema.Schema {
 	}
 }
 
+// expression is optional because it cannot be included in composite alerts but is otherwise required
```
Comment (lambcode) [inline lightstep/resource_metric_condition.go:315]: I'm not sure I follow this comment. There doesn't appear to be anything marked optional here.

**Review by lambcode — COMMENTED** [review]
Sorry for the late review, but I have a few questions on this one.

**Pull request merged**